_overviews/scala3-book/control-structures.md

```diff
@@ -236,7 +236,7 @@ for i <- ints do println(i)
 {% endtabs %}
 
 
-The code `i <- ints` is referred to as a _generator_.
+The code `i <- ints` is referred to as a [_generator_](https://en.wikipedia.org/wiki/Generator_(computer_programming)).
```

Comment:
really the only thing "generator" means in scala is the syntax `p <- e` itself. That wikipedia definition would be inappropriate in this case

Reply:
so the definition of generator is `p <- e` where `e` is an expression, and `p` is a pattern

Reply:
a for-comprehension itself (i.e. the whole `for (exprs) yield e` fulfils the wikipedia generator definition better

Reply:
this is the definition in [the spec](https://scala-lang.org/files/archive/spec/3.4/06-expressions.html#for-comprehensions-and-for-loops)

> A generator `p <- e` produces bindings from an expression `e` which is matched in some way against pattern `p`.